test/foundry/fork/base/TestNexusSwapETHIntegration.t.sol

```diff
@@ -0,0 +1,199 @@
+        uint256 gasUsed = initialGas - gasleft();
+        emit log_named_uint("UniswapV2::swapExactETHForTokens::Gas used for swapping ETH for USDC (EOA)", gasUsed);
+
+        assertGt(usdc.balanceOf(swapper), initialTokenBalance);
+        vm.stopPrank();
+    }
+
+    /// @notice Tests gas consumption for swapping ETH for USDC using a deployed Nexus account
+    function test_Gas_Swap_DeployedNexus_SwapEthForTokens() public {
+        Nexus deployedNexus = deployNexus(user, 100 ether, address(VALIDATOR_MODULE));
+        uint256 initialGas = gasleft();
+
+        Execution[] memory executions = prepareSingleExecution(
+            address(uniswapV2Router),
+            SWAP_AMOUNT,
+            abi.encodeWithSignature(
+                "swapExactETHForTokens(uint256,address[],address,uint256)",
+                0,
+                getPathForETHtoUSDC(),
+                address(deployedNexus),
+                block.timestamp
+            )
+        );
+
+        PackedUserOperation[] memory userOps = buildPackedUserOperation(user, deployedNexus, EXECTYPE_DEFAULT, executions, address(VALIDATOR_MODULE));
+
+        ENTRYPOINT.handleOps(userOps, payable(BUNDLER.addr));
+        uint256 gasUsed = initialGas - gasleft();
+        emit log_named_uint("UniswapV2::swapExactETHForTokens::Gas used for swapping ETH for USDC (Deployed Nexus)", gasUsed);
+
+        assertGt(usdc.balanceOf(address(deployedNexus)), 0);
+    }
+
+    /// @notice Tests deploying Nexus and swapping ETH for USDC with Paymaster
+    function test_Gas_Swap_DeployAndSwap_WithPaymaster() public {
+        uint256 initialGas = gasleft();
+
+        Execution[] memory executions = prepareSingleExecution(
+            address(uniswapV2Router),
+            SWAP_AMOUNT,
+            abi.encodeWithSignature(
+                "swapExactETHForTokens(uint256,address[],address,uint256)",
+                0,
+                getPathForETHtoUSDC(),
+                preComputedAddress,
+                block.timestamp
+            )
+        );
+
+        PackedUserOperation[] memory userOps = buildPackedUserOperation(
+            user,
+            Nexus(preComputedAddress),
+            EXECTYPE_DEFAULT,
+            executions,
+            address(VALIDATOR_MODULE)
+        );
+
+        userOps[0].initCode = buildInitCode(user.addr, address(VALIDATOR_MODULE));
+
+        // Including paymaster address and additional data
+        userOps[0].paymasterAndData = abi.encodePacked(
+            address(paymaster),
+            uint128(3e6), // verification gas limit
+            uint128(3e6) // postOp gas limit
+        );
+
+        userOps[0].signature = signUserOp(user, userOps[0]);
+
+        ENTRYPOINT.handleOps(userOps, BUNDLER_ADDRESS);
+        uint256 gasUsed = initialGas - gasleft();
+        emit log_named_uint("UniswapV2::swapExactETHForTokens::Gas used for deploying Nexus and swapping ETH for USDC with Paymaster", gasUsed);
```

Comment:
for all of these gas values, add 21000 + calldata cost as discussed 

you can use these
https://github.com/rhinestonewtf/modulekit/blob/main/src/test/utils/GasCalculations.sol#L23